src/Core/Util/SymfonyArrayFinder.php

```diff
@@ -0,0 +1,146 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Util;
+
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessor;
+
+class SymfonyArrayFinder implements \ArrayAccess, \Countable
+{
+    /**
+     * @var array
+     */
+    private $array;
+    /**
+     * @var PropertyAccessor
+     */
+    private $propertyAccessor;
+
+    public function __construct(array $content = [])
+    {
+        $this->array = $content;
+        $this->propertyAccessor = PropertyAccess::createPropertyAccessor();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function count()
+    {
+        return count($this->array);
+    }
+
+    private function convertDotPathToArrayPath(string $dotPath): string
```

Comment:
Please sort your methods by visibility: first public, then protected, then private

Reply:
I consider it a good practice to describe what your method does, its inputs and outputs so that people can get a sense of what it does without having to reverse engineer the code

Reply:
true, and a well-named function is better than a comment tediously explaining what it does

Reply:
Woups!
I always sort my methods but someone slipped through 😫

Reply:
Done in https://github.com/PrestaShop/PrestaShop/pull/25611/commits/c0066eb57f7c1f3ed17205f820cc4a0a789c5073